PyAssembler/Assembler.py

```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/env python3.7
+
+import sys
+import struct
+import argparse
+import binascii
+from enum import Enum
+
+
+STR_R_FORMAT = 'r'
+BYTES_W_FORMAT = 'wb'
+
+
+class Assembler:
+
+    class TooManyParamsException(TypeError):
+        """
+        Raised when the input line consist of 
+        more than 2 parameters """
+
+        def __init__(self, code_line):
+            self.code_line = code_line
+
+        def __str__(self):
+            return 'Invalid assembly code line: Too many arguments at line: {0} '.format(self.code_line)

```

Comment:
**TIP:** If you're not gonna reorder the arguments or print one of them twice, you can omit all numbers from the curly braces.

Example:
```python
print('hello {}!'.format('John'))
```